pkg/autoscaler/aggregation/max/window.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package max
+
+import (
+	"fmt"
+
+	"github.com/davecgh/go-spew/spew"
+)
+
+type entry struct {
+	value float64
+	index int
+}
+
+// Window is a circular buffer which keeps track of the maximum value observed in a particular time.
+// Based on the "ascending minima algorithm" (http://web.archive.org/web/20120805114719/http://home.tiac.net/~cri/2001/slidingmin.html).
+type Window struct {
```

Comment:
should this (and all the functions in here) be package private?

Reply:
torn, but, making it public later is easier than making it private once people have started using it, so lets err on the side of the more reversible decision - updated.